packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -186,14 +186,24 @@ export default class PerformanceMonitoring {
         task.eventType === 'click'
       ) {
         const target = task.target
-        const name = target.getAttribute('name')
+        const name =
+          typeof target.getAttribute === 'function'
+            ? target.getAttribute('name')
+            : ''
```

Comment:
This logic here should also be applied to a line further down in this file, [click here (Line 224)](https://github.com/elastic/apm-agent-rum-js/blob/cd722fefc10dde2c29626144083e7b7a82477a97/packages/rum-core/src/performance-monitoring/performance-monitoring.js#L224). Otherwise we'll see a `TypeError: target.getAttribute is not a function` when the function does not exist.

Reply:
You are right, `document` and `window` have no `getAttribute`.

Reply:
@vigneshshanmugam Should we change `typeof xxx === 'function'` to `isFunction(xxx)` to save some bytes?
```bash
$ ag -c "'function'"
common/config-service.js:1
common/patching/event-target-patch.js:1
common/patching/history-patch.js:1
common/patching/patch-utils.js:1
common/logging-service.js:1
common/service-factory.js:1
common/event-handler.js:1
common/utils.js:10
common/compress.js:1
performance-monitoring/performance-monitoring.js:2
performance-monitoring/span-base.js:1
common/throttle.js:1
performance-monitoring/transaction-service.js:1
error-logging/error-logging.js:3
```

Reply:
Hmmm might not give any advantage when its gzipped, did you verify?

Reply:
Before gzipped, saved 353 bytes: 57792-57439
After gzipped, saved only 5 bytes: 18808-18803